packages/suite/src/reducers/suite/messageSystemReducer.ts

```diff
@@ -100,47 +100,43 @@ const messageSystemReducer = (
-        (state: MessageSystemRootState) => state.messageSystem.dismissedMessages,
-        (config, validMessages, dismissedMessages) => {
-            const nonDismissedValidMessages = validMessages[category].filter(
-                id => !dismissedMessages[id]?.[category],
-            );
+    memoize((state: MessageSystemRootState) => {
+        const { config, validMessages, dismissedMessages } = state.messageSystem;
+        const nonDismissedValidMessages = validMessages[category].filter(
+            id => !dismissedMessages[id]?.[category],
+        );
 
-            const messages = config?.actions
-                .filter(({ message }) => nonDismissedValidMessages.includes(message.id))
-                .map(action => action.message);
+        const messages = config?.actions
+            .filter(({ message }) => nonDismissedValidMessages.includes(message.id))
+            .map(action => action.message);
 
-            if (!messages?.length) return [];
+        if (!messages?.length) return [];
 
-            return messages.sort(comparePriority);
-        },
-    );
+        return messages.sort(comparePriority);
+    });
 
 export const selectActiveBannerMessages = makeSelectActiveMessagesByCategory('banner');
 export const selectActiveContextMessages = makeSelectActiveMessagesByCategory('context');
 export const selectActiveModalMessages = makeSelectActiveMessagesByCategory('modal');
 export const selectActiveFeatureMessages = makeSelectActiveMessagesByCategory('feature');
 
-export const selectBannerMessage = createSelector(
-    selectActiveBannerMessages,
-    messages => messages[0],
-);
+export const selectBannerMessage = memoize((state: MessageSystemRootState) => {
+    const activeBannerMessages = selectActiveBannerMessages(state);
+    return activeBannerMessages[0];
+});
 
-export const selectContextMessage = createSelector(
-    selectActiveContextMessages,
-    (_state: MessageSystemRootState, domain: ContextDomain) => domain,
-    (messages, domain) => messages.find(message => message.context?.domain === domain),
+export const selectContextMessage = memoizeWithArgs(
+    (state: MessageSystemRootState, domain: ContextDomain) => {
+        const activeContextMessages = selectActiveContextMessages(state);
+        return activeContextMessages.find(message => message.context?.domain === domain);
+    },
 );
 
-export const selectContextMessageContent = createSelector(
-    selectActiveContextMessages,
-    (_state: MessageSystemRootState, domain: ContextDomain) => domain,
-    selectLanguage,
-    (messages, domain, language) => {
-        const message = messages.find(message => message.context?.domain === domain);
+export const selectContextMessageContent = memoizeWithArgs(
+    (state: MessageSystemRootState, domain: ContextDomain) => {
+        const activeContextMessages = selectActiveContextMessages(state);
+        const language = selectLanguage(state as any);
```

Comment:
What about this to avoid `as any`?
```suggestion
    (state: MessageSystemRootState & SuiteRootState, domain: ContextDomain) => {
        const activeContextMessages = selectActiveContextMessages(state);
        const language = selectLanguage(state);
```

Reply:
I added upper to `MessageSystemRootState` because it need this part of "root state" to work. https://github.com/trezor/trezor-suite/pull/7413/commits/c1b7400a238e9606f2d10d1e916108f409ddd360